include/net/ieee802154_radio.h

```diff
@@ -171,8 +172,8 @@ enum ieee802154_config_type {
 	/** Configure a radio reception slot */
 	IEEE802154_CONFIG_RX_SLOT,
 
-	/** Enable CSL receiver (Endpoint) */
-	IEEE802154_CONFIG_CSL_RECEIVER,
+	/** Configure CSL receiver (Endpoint) period */
+	IEEE802154_CONFIG_CSL_PERIOD,
```

Comment:
Does that mean there's no runtime configuration for enabling/disabling CLS? Just a compile-time switch?

Reply:
In current API design, radio drivers do not support CSL. They just expose function that enable higher layers to implement CSL. Because of this, it is not possible to enable/disable CSL runtime.

If higher layer would like to disable CSL, it should just stop calling `config(IEEE802154_CONFIG_CSL_RX_TIME, ...)` and reset Header IEs in Enh-Acks.

Reply:
From what I understand, in the case of OpenThread, if we build with the CSL support we can turn it on/off in runtime.

Reply:
From IEEE 802.15.4-2-15:

> The CSL mode is turned on when the PIB attribute macCslPeriod is nonzero and turned off when macCslPeriod is zero.

Reply:
> From IEEE 802.15.4-2015:
> 
> > The CSL mode is turned on when the PIB attribute macCslPeriod is nonzero and turned off when macCslPeriod is zero.

But it does not work for this API. If radio drivers fully supported CSL, you would need this simple function to set `IEEE802154_CONFIG_CSL_PERIOD` to enable or disable CSL. However, in current design, most of CSL is implemented in a higher (MAC) layer that must keep setting `IEEE802154_CONFIG_RX_SLOT`, `IEEE802154_CONFIG_CSL_RX_TIME` sequentially to keep CSL active.

Reply:
> But it does not work for this API. If radio drivers fully supported CSL, you would need this simple function to set IEEE802154_CONFIG_CSL_PERIOD to enable or disable CSL.

I think I'm understanding even less about the feature now. What about the `nrf5_config_csl_period()` implementation then, it seems to call `nrf5_receive_at()` underneath - what's the point if the upper layer is supposed to call `IEEE802154_CONFIG_CSL_RX_TIME `/`IEEE802154_CONFIG_RX_SLOT` anyway? I think we need better documentation for these configuration options given that we're confused about the usage even internally.

Reply:
@rlubos do you think this point should be better explained?

```
/* A placeholder reception window is scheduled so that the radio driver is able to inject
 * the proper CSL Phase in the transmitted CSL Information Elements.
 */
```

Reply:
@edmont I wonder if `IEEE802154_CONFIG_CSL_PERIOD` should have a side effect of setting an RX slot, or whether the user should rather call `IEEE802154_CONFIG_CSL_PERIOD`/`IEEE802154_CONFIG_RX_SLOT` slot combo. 

But if it indeed should have configure the RX slot as well (I assume it's a single shot, as from the discussion above it seems the radio driver will not reschedule the next slot) this information should be at very least documented in the `ieee802154_radio.h` option descripiton, not hidden inside Nordic implementation.

Reply:
I agree we should document it better. Right now implementing CSL in higher layer is enigmatic.

Basically, to implement CSL higher layer must:

1. Set `IEEE802154_CONFIG_CSL_PERIOD`
2. Set following (not the closes one) CSL window time (`IEEE802154_CONFIG_CSL_RX_TIME`)
3. Schedule the closest CSL window (set `IEEE802154_CONFIG_RX_SLOT`)
4. After the closest CSL window is finished, go to step 2

When period is changed, the higher layer shall:

1. Set new `IEEE802154_CONFIG_CSL_PERIOD`
2. Continue iteration from step 2 above

Reply:
From a driver implementation point of view:

`IEEE802154_CONFIG_CSL_PERIOD` is needed to set period in CSL IE
`IEEE802154_CONFIG_CSL_RX_TIME` is needed to set phase in CSL IE
`IEEE802154_CONFIG_RX_SLOT` is needed to schedule RX window precisely

Reply:
Thanks for explaination. Shouldn't `IEEE802154_CONFIG_CSL_RX_TIME` be confgured to some value before `IEEE802154_CONFIG_CSL_PERIOD` though, given it makes use `nrf5_data.csl_rx_time`?

Anyway, I am aware we're tight on schedule, so don't bother with the documentation now if the solution works for you. But we should really consider improving it when submitting the patches upstream.

Reply:
> Shouldn't IEEE802154_CONFIG_CSL_RX_TIME be confgured to some value before IEEE802154_CONFIG_CSL_PERIOD though, given it makes use nrf5_data.csl_rx_time?

Right. Fixing in the comment above.